apps/yapms/src/lib/components/modals/sharemodal/ShareModal.svelte

```diff
@@ -99,13 +102,32 @@
 	function onTurnstileExpired() {
 		turnstileToken = null;
 	}
+
+	async function screenshot() {
+		const mapChart = document.getElementById('map-chart-div');
+		if (mapChart === null) {
+			return;
+		}
+
+		const data = await htmlToImage.toBlob(mapChart, {
+			backgroundColor: 'hsl(var(--b1))'
+		});
+
+		if (data !== null) {
+			fileSaver.saveAs(data, 'screenshot.png');
```

Comment:
Nit: maybe we make this YAPmsScreenshot like YAPmsMap?

Reply:
Changed